src/include/optimizer/statistics/stats_storage.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include "catalog/catalog_defs.h"
+#include "common/hash_util.h"
+#include "common/macros.h"
+#include "common/managed_pointer.h"
+
+#include "optimizer/statistics/column_stats.h"
+#include "optimizer/statistics/table_stats.h"
+
+#include <sstream>
+
+namespace terrier::optimizer {
+
+/**
+ * An object made to be hashable so it can be a key to the table stats
+ * storage map.
+ */
+class StatsStorageKey {
```

Comment:
Creating a new class to store two integers seems like an overkill. Why not just `typedef` on a `std::pair`?